xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -0,0 +1,737 @@
+          // immediately, so execute handleResolvedAddresses() after all the setup in this method is
+          // complete.
+          helper.getSynchronizationContext().execute(new Runnable() {
+            @Override
+            public void run() {
+              localityLbInfo.childBalancer.handleResolvedAddresses(
+                  ResolvedAddresses.newBuilder().setAddresses(eags).build());
+            }
+          });
+        }
+      }
+
+      for (Locality newLocality : newLocalities) {
+        if (!localities.contains(newLocality)) {
+          loadStatsStore.addLocality(newLocality);
+        }
+      }
+      final Set<Locality> toBeRemovedFromStatsStore = new HashSet<>();
+      // There is a race between picking a subchannel and updating localities, which leads to
+      // the possibility that RPCs will be sent to a removed locality. As a result, those RPC
+      // loads will not be recorded. We consider this to be natural. By removing locality counters
+      // after updating subchannel pickers, we eliminate the race and conservatively record loads
+      // happening in that period.
+      for (Locality oldLocality : localities) {
+        if (!localityInfoMap.containsKey(oldLocality)) {
+          toBeRemovedFromStatsStore.add(oldLocality);
+        }
+      }
+      helper.getSynchronizationContext().execute(new Runnable() {
+        @Override
+        public void run() {
+          for (Locality locality : toBeRemovedFromStatsStore) {
+            loadStatsStore.removeLocality(locality);
+          }
+        }
+      });
+      localities = newLocalities;
+
+      priorityManager.updateLocalities(localityInfoMap);
+
+      for (Locality oldLocality : localityMap.keySet()) {
+        if (!newLocalities.contains(oldLocality)) {
+          deactivate(oldLocality);
+        }
+      }
+    }
+
+    private void updateDropPercentage(List<DropOverload> dropOverloads) {
+      this.dropOverloads = checkNotNull(dropOverloads, "dropOverloads");
+    }
+
+    private void deactivate(final Locality locality) {
+      if (!localityMap.containsKey(locality) || localityMap.get(locality).isDeactivated()) {
+        return;
+      }
+
+      final LocalityLbInfo localityLbInfo = localityMap.get(locality);
+      class DeletionTask implements Runnable {
+
+        @Override
+        public void run() {
+          localityLbInfo.shutdown();
+          localityMap.remove(locality);
+        }
+
+        @Override
+        public String toString() {
+          return "DeletionTask: locality=" + locality;
+        }
+      }
+
+      localityLbInfo.delayedDeletionTimer = helper.getSynchronizationContext().schedule(
+          new DeletionTask(), DELAYED_DELETION_TIMEOUT_MINUTES,
+          TimeUnit.MINUTES, helper.getScheduledExecutorService());
+    }
+
+    // TODO(zdapeng): Decide how to work with LRS client.
+    LoadStatsStore getLoadStatsStore() {
```

Comment:
Very likely no longer need this method. We should figure out how LRS to be integrated/initiated.